admin/class-acc_user_importer-admin.php

```diff
@@ -19,84 +163,139 @@ public function __construct( $plugin_name, $version ) {
+	// Get the section name as per the settings
+	private function getSectionName ( ) {
+		$options = get_option('accUM_data');
+		if (!isset($options['accUM_section_api_id'])) {
+			$sectionName = accUM_get_section_default();
+		} else {
+			$sectionName = $options['accUM_section_api_id'];
+		}
+		return $sectionName;
+	}
 
-		//force certificate validation - i.e. speed up authentication process
-		add_filter( 'https_local_ssl_verify', '__return_true' );
+	// Get the section API ID
+	private function getSectionApiID ( $sectionName ) {
+		return($this->sectionApiId[$sectionName]);
+	}
 
+	/*
+	 * Returns the token for the section the plugin is operating on.
+	 * Null is returned if no token has been defined for the section.
+	 * Token format: the section name must be as in the accUM_section_api_id list.
+	 * Do not put spaces before or after the comma. Here are valid examples
+	 * (with bogus token values):
+	 * 		SQUAMISH:fpZloKQj8L
+	 * 		COLUMBIA MOUNTAINS:123,MONTRÉAL:666,OUTAOUAIS:HJD634
+	 */
+	private function getSectionToken() {
 		$options = get_option('accUM_data');
-		$acc_user = $options['accUM_username'];
-		$this->log_local_output("Automatic member update starting for section $acc_user");
-		$timestamp_start = date_i18n("Y-m-d-H-i-s");
+		$sectionName = $this->getSectionName();
+		$tokenStrings = explode(',', $options['accUM_token']);
+
+		foreach ($tokenStrings as $tokenString) {
+			$sectionEntry = [];
+			$sectionTokenStrings = explode(':', $tokenString);
+			if (count($sectionTokenStrings) != 2) {
+				$this->log_dual("Error, each section token should have 2 params");
+				return null;
+			}
+			if ($sectionTokenStrings[0] == $sectionName) {
+				// Found the token for the current section
+				//$this->log_dual("Token for section {$sectionName} is {$sectionTokenStrings[1]}");
+				return $sectionTokenStrings[1];
+			}
+		}
+
+		$this->log_dual("Error, no token provided for section {$sectionName}");
+		return null;
+	}
 
-		//request token
-		$this->log_local_output("Requesting access token from national office.");
-		$access_token_request = $this->request_API_token();
 
-		//did we get token?
-		if ( $access_token_request['message'] == "success") {
+	public function responseErrMsg($api_response) {
+		return("Error: " . ($api_response['errorMessage'] ? $api_response['errorMessage'] : 'Unknown.'));
+	}
 
-			$this->log_local_output('Received token: ' . substr($access_token_request['accessToken'], 0, 10) . ".");
 
-			$has_next = true;
-			$get_attempts_remaining = 3;
-			$data_offset = 0;
+	/**
+	 * This is the user import loop (when triggered by a timer)
+	 */
+	public function begin_automatic_update() {
+
+		$GLOBALS['acc_logstr'] = "";		//Clear the API response log string
 
-			//get data until no more data exists
-			while ( ($has_next === true) && ($get_attempts_remaining > 0) ) {
+		//force certificate validation - i.e. speed up authentication process
+		add_filter( 'https_local_ssl_verify', '__return_true' );
```

Comment:
I am generally not a big fan of skipping SSL verification, how much overhead did this add?